idn/punycode.go

```diff
@@ -52,6 +59,12 @@ func ToPunycode(s string) string {
 
 // FromPunycode returns unicode domain name from provided punycode string.
 func FromPunycode(s string) string {
+	// Early check to see if decoding is needed.
+	// This will prevent making heap allocations when not needed.
+	if !strings.Contains(s, _PREFIX) {
```

Comment:
That won't work since it can be in any of the labels. For example `www.xn--z-0ub.com`.